app/frontend/src/components/forms/manage/ManageVersions.vue

```diff
@@ -0,0 +1,208 @@
+              color="primary"
+              text
+              small
+              @click="exportSchema(item.id)"
+              v-bind="attrs"
+              v-on="on"
+            >
+              <v-icon>get_app</v-icon>
+            </v-btn>
+          </template>
+          <span>Download version {{ item.version }} to a file</span>
+        </v-tooltip>
+      </template>
+    </v-data-table>
+
+    <BaseDialog
+      v-model="showHasDraftsDialog"
+      type="OK"
+      @close-dialog="showHasDraftsDialog = false"
+    >
+      <template #title>Draft already exists</template>
+      <template #text>
+        A Draft already exists for this form. Please edit that draft or delete
+        it if you wish to start a new one.
+      </template>
+    </BaseDialog>
+  </div>
+</template>
+
+<script>
+import { mapActions, mapGetters } from 'vuex';
+
+import { FormPermissions } from '@/utils/constants';
+import CurrentDraft from '@/components/forms/manage/CurrentDraft.vue';
+import formService from '@/services/formService.js';
+
+export default {
+  name: 'ManageVersions',
+  components: { CurrentDraft },
+  data() {
+    return {
+      headers: [
+        { text: 'Version', align: 'start', value: 'version' },
+        { text: 'Date Published', align: 'start', value: 'createdAt' },
+        {
+          text: 'Create a New Version',
+          align: 'center',
+          value: 'create',
+          filterable: false,
+          sortable: false,
+        },
+        {
+          text: 'Export',
+          align: 'center',
+          value: 'export',
+          filterable: false,
+          sortable: false,
+        },
+      ],
+      showHasDraftsDialog: false,
+    };
+  },
+  computed: {
+    ...mapGetters('form', ['drafts', 'form', 'permissions']),
+    canCreateDesign() {
+      return this.permissions.includes(FormPermissions.DESIGN_CREATE);
+    },
+    hasDraft() {
+      return this.drafts && this.drafts.length;
+    },
+    versionList() {
+      return this.form ? this.form.versions : [];
+    },
+  },
+  methods: {
+    ...mapActions('notifications', ['addNotification']),
+    createVersion(formId, versionId) {
+      if (this.hasDraft) {
+        this.showHasDraftsDialog = true;
+      } else {
+        this.$router.push({
+          name: 'FormDesigner',
+          query: { f: formId, v: versionId },
+        });
+      }
+    },
+    async exportSchema(versionId) {
+      try {
+        const ver = await formService.readVersion(this.form.id, versionId);
+        if (ver && ver.data) {
+          const a = document.createElement('a');
+          a.href = `data:application/json;charset=utf-8,${encodeURIComponent(
+            JSON.stringify(ver.data.schema)
+          )}`;
+          a.download = `${this.form.snake}_schema.json`;
+          a.style.display = 'none';
+          a.classList.add('hiddenDownloadTextElement');
+          document.body.appendChild(a);
+          a.click();
+          document.body.removeChild(a);
```

Comment:
I think we have this logic appearing in a few places in our code. It may be worth consolidating this into a utils-style function down the line to reduce duplication.

Reply:
Yeah should look at that/. I think I tried a while back but ran out of time trying to make it work